exercises/practice/pascals-triangle/.meta/lib/example.dart

```diff
@@ -3,18 +3,23 @@ class PascalsTriangle {
     if (rows <= 0) {
       return [];
     } else {
-      List<List<int>> triangle = new List<List<int>>();
+      final triangle = <List<int>>[];
 
       for (int x = 1; x <= rows; x++) {
-        List<int> nextRow = new List<int>(x)
-          ..[0] = 1
-          ..[x - 1] = 1;
+        final nextRow = <int>[];
+
         if (triangle.length > 1) {
-          List<int> previousRow = triangle.elementAt(triangle.length - 1);
+          nextRow.add(1);
```

Comment:
This is what I could do to pass the tests. As of 2.12, the default constructors for the collections and map are deprecated.

Instead they want you to either use the collection literal or one of the named constructors.

So that means, for a set, I cannot set the size of it and then only set the value of the first and last indexes, and then fill out in between.

From my understanding, which can be corrected, it forced me to go in sequential order.

Reply:
I wrote another approach for this example, that makes things a little bit more explicit. This passes all tests
It's short and keeps some information about the indexes. It looks as if we are writing all of the rows by hand in list literals.
What do you think?
```dart
class PascalsTriangle {
  List<List<int>> rows(int rows) {
    /// Row that was last inserted in the triangle
    /// This also add statefull behaviour to the closure passed to List.generate
    List<int> lastRow = [];

    return List.generate(rows, (rowIdx) {
      return lastRow = [
        /// Leading 1
        1,

        /// Expand on lastRow, only if it's length is greater than 1
        if (lastRow.length > 1)
          for (int i = 1; i < lastRow.length; i++) lastRow[i - 1] + lastRow[i],

        /// Trailing 1, only if this row's length is greater than 1
        if (rowIdx > 0) 1,
      ];
    });
  }
}
```

Reply:
Thanks @devkabiir. I made a few changes to the comments and renamed lastRow to triangle to better reflect what was being returned. Additionally, I separated the assignment from the return within generate's function in order to make the code easier to read.